Unify payment method icon design

KarlisJ opened the pull request:
Fixes #7183

#### Changes proposed in this Pull Request

Unify the design for Payment method and Card icon design.
- borders are consistent 
- alignement is consistent

#### Testing instructions

TODO: Need to list all use places. Quite hard to gather them all

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Commit:
Stash changes for review

bradyv committed:
Fix border on payment method icons

Commit:
Add new icons to settings

Commit:
Allow making icons smallers

Commit:
Unify icons borders

KarlisJ commented:
We still need to fix some failing tests, mostly just recreating snapshots.

Also, this PR needs to address the use cases where assets are used as CSS backgrounds. I might skip it since there are no problems with alignment or borders.

kalessil commented on client/payment-methods-icons.tsx:
```diff
@@ -0,0 +1,80 @@
+	props
+) => <img className={ 'payment-method__icon' } src={ src } alt={ alt } />;
+
+// export const CreditCardIcon = iconComponent(
```
Dead code?

kalessil commented on client/payment-methods-icons.tsx:
```diff
@@ -0,0 +1,80 @@
+// 	'Credit / Debit card'
+// );
+
+export const BancontactIcon = iconComponent( BancontactAsset, 'Bancontact' );
```
Not sure, but we should probably make alt-text translatable for all icons declared here.

kalessil commented on client/payment-methods-icons.tsx:
```diff
@@ -0,0 +1,80 @@
+	'American Express'
+);
+export const DinersClubIcon = iconComponent( DinersClubAsset, 'Diners Club' );
+export const UnionPayIcon = iconComponent( UnionPayAsset, 'UnionPay' );
```
It'd also be great to have constants sorted by name.

kalessil reviewed (commented):
Early feedback: looks good! Leaving some comments to look into (minor ones).

Commit:
Sort icons alphabetically

Commit:
Make alt properties translatable

mgascam commented on client/payment-methods-map.tsx:
```diff
@@ -200,11 +197,10 @@ const PaymentMethodInformationObject: Record<
 		description: __(
-			// translators: %s is the store currency.
-			'Allow customers to pay over time with Affirm. Available to all customers paying in %s.',
+			'Expand your business with Affirm',
```
I noticed we are changing the description for Affirm and Afterpay, is this intended? Asking because for Klarna we are not changing this description.

KarlisJ replied:
Not intended, thanks for pointing out. Could be incorrect merge from `develop`. I'll look into that.

mgascam commented on client/style.scss:
```diff
@@ -169,3 +169,24 @@
+
+img {
+	&.payment-method__icon {
+		outline: 1px solid rgba( 0, 0, 0, 0.25 );
```
out of curiosity, why we use `outline` instead of `border` ? Asking because border is more commonly used.

KarlisJ replied:
@bradyv Could answer better to this question as the credit for CSS goes to him.

AFAIU, the issue with the border was that it's is drawn around the asset which created the inconsistencies between CC icon (which had the individual borders baked into the asset) and other assets. 

With the new grid approach that might not be the problem anymore. 

Luckily, we now have single source of truth for the border CSS and it can be easily adjusted.

bradyv replied:
@mgascam Basically exactly what @KarlisJ  said. `outline` allows us to set a negative offset that draws the stroke on top of the icon. Using `border` required additional mockup that made it much more difficult to scale to different icon sizes.

Beyond just inconsistencies between assets this also gives us a much nicer display as the stroke is clearly visible for all icons regardless of icon colour or the background colour they're displayed on.

mgascam replied:
Thanks for clarification @bradyv @KarlisJ

mgascam commented on client/payment-methods-map.tsx:
```diff
@@ -217,11 +213,10 @@ const PaymentMethodInformationObject: Record<
 		description: __(
-			// translators: %s is the store currency.
-			'Allow customers to pay over time with Afterpay. Available to all customers paying in %s.',
+			'Expand your business with Afterpay',
```
Same as previous comment

mgascam reviewed (commented):
Looking great already! I left a couple of comments to double check with you.

One thing I noticed is that the Link icon does not have border on the Payment > Settings screen in the Express Checkout section.

<img width="688" alt="Screenshot 2023-10-26 at 11 48 26" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/4680b48d-f30f-47e2-80e8-791c56aecc7e">

kalessil commented on client/payment-methods-icons.tsx:
```diff
@@ -0,0 +1,84 @@
+	props
+) => <img className={ 'payment-method__icon' } src={ src } alt={ alt } />;
+
+export const AffirmIcon = iconComponent( AffirmAsset, __( 'Affirm' ) );
```
It's typically 2 parameters: string and the domain `woocommerce-payments`

Commit:
Fix: add domain to localised alt

Commit:
Add LinkIcon

Commit:
Apply unified border for icons in dispute/transaction tables

Commit:
Revert incorrect merge changes

KarlisJ committed:
Merge branch 'develop' into fix/7183-unify-pm-icon-design

KarlisJ commented:
@mgascam Thank you for pointing out the missing LinkIcon.

I've implemented all requested changes. Just need to resolve failing tests (in most cases recreating snapshot)

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7560` or branch name `fix/7183-unify-pm-icon-design` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7183-unify-pm-icon-design) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 86a81f17cc693ba27488490b1f2ecfc4b537bdd2
- Build time: 2023-10-27 13:15:32 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Commit:
Add changelog

dmvrtx reviewed (approved):
Everything looks good and consistent on the Settings and Transactions screens! Good work!

Commit:
allow passing className to the root img tag

Some use cases like removing PM modal relies on additional classes added to the root img

Commit:
Use single CC asset instead of grid

Current grid component is not resiable and brakes the design for enabling Credit card modal

Commit:
Fix tests

KarlisJ committed:
Merge branch 'develop' into fix/7183-unify-pm-icon-design
